xarray/tests/test_interp.py

```diff
@@ -245,38 +282,39 @@ def func(obj, dim, new_x, method):
     assert_allclose(actual, expected.transpose("z", "w", "y", transpose_coords=True))
 
 
+@requires_scipy
+@pytest.mark.parametrize("method", ("linear", "nearest", "slinear", "pchip"))
 @pytest.mark.parametrize(
     "case", [pytest.param(3, id="no_chunk"), pytest.param(4, id="chunked")]
 )
-def test_interpolate_nd(case: int) -> None:
-    if not has_scipy:
-        pytest.skip("scipy is not installed.")
-
+def test_interpolate_nd_separable(
+    case: int, method: InterpnOptions, nd_interp_coords
+) -> None:
     if not has_dask and case == 4:
         pytest.skip("dask is not installed in the environment.")
 
     da = get_example_data(case)
 
     # grid -> grid
-    xdestnp = np.linspace(0.1, 1.0, 11)
-    ydestnp = np.linspace(0.0, 0.2, 10)
-    actual = da.interp(x=xdestnp, y=ydestnp, method="linear")
+    xdestnp = nd_interp_coords["xdestnp"]
+    ydestnp = nd_interp_coords["ydestnp"]
+    actual = da.interp(x=xdestnp, y=ydestnp, method=method)
 
-    # linear interpolation is separateable
-    expected = da.interp(x=xdestnp, method="linear")
-    expected = expected.interp(y=ydestnp, method="linear")
+    # `method` is separable
```

Comment:
is `pchip` really separable too? We could gain some confidence here by constructing expected using `scipy` directly.

Reply:
My experiments suggest only cubic and quintic (or other splines with order > 1) are "truly multidimensional." I'm interpreting errors on the order of 1e-16 relative to the SD of the expected function as machine error, but errors on the order of 1e-5 as true errors. 

It feels a bit silly to operate "empirically" like this, but digging through the formalism behind these interpolants would be a big lift and I'm not able to take it on right now. I sort of wish the separation optimization didn't exist and we could just provide a transparent interface to SciPy, but I guess this is insufficiently considerate to the user. 

```python3
import scipy
import xarray as xr
import numpy as np 
import matplotlib.pyplot as plt
import seaborn as sb

orig_x = np.linspace(0,2,10)
orig_y = np.linspace(0,1,10)
orig_f = np.random.randn(10,10)

fine_x = np.linspace(0,2,200)
fine_y = np.linspace(0,1,100)
mesh_x, mesh_y = np.meshgrid(fine_x, fine_y)
all_points = np.asarray([x for x in zip(mesh_x.ravel(), mesh_y.ravel())])
new_x_da = xr.DataArray(data = np.transpose(mesh_x), dims = ["x","y"], coords = {"x": fine_x, "y": fine_y})
new_y_da = xr.DataArray(data = np.transpose(mesh_y), dims = ["x","y"], coords = {"x": fine_x, "y": fine_y})

mtds = ["linear", "nearest", "slinear", "cubic", "quintic", "pchip"]
for mtd in mtds:
    result_interpn = scipy.interpolate.interpn(points = (orig_x, orig_y), values = orig_f, xi = all_points, method = mtd).reshape(200,100, order="F")
    result_rgi = scipy.interpolate.RegularGridInterpolator(points = (orig_x, orig_y), values = orig_f, method= mtd)(all_points).reshape(200,100, order="F")
    result_xr = xr.DataArray(data = orig_f, dims = ["x","y"], coords = {"x": orig_x, "y": orig_y}).interp(method= mtd, x = new_x_da, y = new_y_da)
    
    if mtd == "pchip":
        result_sequential = scipy.interpolate.PchipInterpolator(orig_y, scipy.interpolate.PchipInterpolator(orig_x, orig_f, axis = 0)(fine_x), axis = 1)(fine_y)
        result_sequential_2 = scipy.interpolate.PchipInterpolator(orig_x, scipy.interpolate.PchipInterpolator(orig_y, orig_f, axis = 1)(fine_y), axis = 0)(fine_x)
    else:
        if mtd == "quintic":
            mtd = 5
        result_sequential = scipy.interpolate.interp1d(orig_y, scipy.interpolate.interp1d(orig_x, orig_f, axis = 0, kind = mtd)(fine_x), axis = 1, kind = mtd)(fine_y)
        result_sequential_2 = scipy.interpolate.interp1d(orig_x, scipy.interpolate.interp1d(orig_y, orig_f, axis = 1, kind = mtd)(fine_y), axis = 0, kind = mtd)(fine_x)


    fig, axs = plt.subplots(nrows = 2, ncols = 2, figsize = [15,12])

    fig.suptitle(f"method = {mtd}")
    sb.heatmap(result_sequential_2-result_interpn, ax = axs[0,0])
    axs[0,0].set_title("sequential versus scipy.interpn")
    
    sb.heatmap(result_sequential_2-result_rgi, ax = axs[1,0])
    axs[1,0].set_title("sequential versus RegularGridInterpolator")
    
    sb.heatmap(result_sequential_2-result_xr, ax = axs[0,1])
    axs[0,1].set_title("sequential versus xarray.DataArray.interp()")
    
    sb.heatmap(result_sequential - result_sequential_2, ax = axs[1,1])
    axs[1,1].set_title("sequential versus reverse-order sequential")
```

Reply:
OK fair, let's just stick with what you have.